lib/rules/indent.js

```diff
@@ -521,540 +393,486 @@ module.exports = {
+                        setDesiredOffset(token, lastAnchor, 1);
+                    }
+                });
+            },
 
-                // Switch is not a 'BlockStatement'
-                const switchIndent = getNodeIndent(node).goodChar;
-                const caseIndent = expectedCaseIndent(node, switchIndent);
-
-                checkNodesIndent(node.cases, caseIndent);
+            TemplateLiteral(node) {
+                const tokens = getTokensAndComments(node);
+                const firstBacktick = sourceCode.getFirstToken(node);
 
+                setDesiredOffsets(tokens.slice(1, -1), firstBacktick, 1);
 
-                checkLastNodeLineIndent(node, switchIndent);
+                node.quasis.slice(1).forEach(quasi => {
+                    setDesiredOffsets(tokens.filter(token => token.range[0] >= quasi.range[0] && token.range[1] <= quasi.range[1]), firstBacktick, 0);
+                });
             },
 
-            SwitchCase(node) {
-
-                // Skip inline cases
-                if (isSingleLineNode(node)) {
-                    return;
-                }
-                const caseIndent = expectedCaseIndent(node);
-
-                checkNodesIndent(node.consequent, caseIndent + indentSize);
+            VariableDeclaration(node) {
+                setDesiredOffsets(getTokensAndComments(node).slice(1), sourceCode.getFirstToken(node), options.VariableDeclarator[node.kind]);
             },
 
-            FunctionDeclaration(node) {
-                if (isSingleLineNode(node)) {
-                    return;
-                }
-                if (options.FunctionDeclaration.parameters === "first" && node.params.length) {
-                    checkNodesIndent(node.params.slice(1), node.params[0].loc.start.column);
-                } else if (options.FunctionDeclaration.parameters !== null) {
-                    checkNodesIndent(node.params, getNodeIndent(node).goodChar + indentSize * options.FunctionDeclaration.parameters);
+            VariableDeclarator(node) {
+                if (node.init) {
+                    ignoreToken(sourceCode.getFirstToken(node.init));
                 }
             },
 
-            FunctionExpression(node) {
-                if (isSingleLineNode(node)) {
-                    return;
-                }
-                if (options.FunctionExpression.parameters === "first" && node.params.length) {
-                    checkNodesIndent(node.params.slice(1), node.params[0].loc.start.column);
-                } else if (options.FunctionExpression.parameters !== null) {
-                    checkNodesIndent(node.params, getNodeIndent(node).goodChar + indentSize * options.FunctionExpression.parameters);
+            "VariableDeclarator:exit"(node) {
+                if (node.parent.declarations.length > 1 && node.parent.declarations[0] === node && node.init) {
+                    new Set(getTokensAndComments(node.init)).forEach((token, _, set) => {
```

Comment:
Very minor thing, but if you declare `const set = new Set(getTokensAndComments(node.init));`, then the `forEach` only needs one parameter.